star-dash/star-dash/GameEngine/Systems/System.swift

```diff
@@ -0,0 +1,20 @@
+//
+//  System.swift
+//  star-dash
+//
+//  Created by Ho Jun Hao on 13/3/24.
+//
+
+import Foundation
+
+protocol System {
+    var isActive: Bool { get set }
+    var dispatcher: EventModifiable { get set }
```

Comment:
What does this `dispatcher` do ah? Does the `dispatcher` represent an object that can receive requests/events and route them to appropriate methods? If so, I think it has a different set of functionalities from an `EventModifiable`. Perhaps a new protocol called `EventDispatcher` can be created.

Reply:
Also, do all systems require a dispatcher? If not, perhaps this property can be optional